src/main/java/controller/BusController.java

```diff
@@ -1,28 +1,94 @@
 package controller;
 
 import db.DatabaseHandler;
-
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Statement;
+import entity.Bus;
+import java.sql.*;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
 
 public class BusController {
-	DatabaseHandler databaseHandler = new DatabaseHandler();
-	Statement statement = databaseHandler.createStatement();
 
-	public void listAllBuses() {
+	public int listAllBuses() {
+		DatabaseHandler databaseHandler = new DatabaseHandler();
+		Statement statement = databaseHandler.createStatement();
+		int id = -1;
+		System.out.println("List of all buses:");
 		try {
+			if (databaseHandler.getDbConnection() != null) {
 			ResultSet resultSetBuses = statement.executeQuery("SELECT * FROM buses");
 			while (resultSetBuses.next()) {
-				int myDriverId = resultSetBuses.getInt("id");
-				String myBusNumber = resultSetBuses.getString("busNumber");
-				float myBusFuel = resultSetBuses.getFloat("fuel");
-				System.out.println(myDriverId + " " + myBusNumber + " " + myBusFuel);
+				id = resultSetBuses.getInt("id");
+				String busNumber = resultSetBuses.getString("busNumber");
+				int driverId = resultSetBuses.getInt("driverId");
+				float fuel = resultSetBuses.getFloat("fuel");
+				Timestamp createdOn = resultSetBuses.getTimestamp("createdOn");
+				Date purchasedOn = resultSetBuses.getDate("purchasedOn");
+
+				Bus newBus = new Bus(id, busNumber, driverId, fuel, createdOn, purchasedOn);
```

Comment:
Are you here creating new local variable for storing found buses?

Reply:
Yes, but only for one purpose - use its toString() method, which actually does not make much sense. It would be better to create object and **return** it.
 I will correct this.